File: src/javademo/PirateWarshipLoginForm.java
package javademo;

import javax.swing.*;
import javax.swing.border.EmptyBorder;
import javax.swing.border.LineBorder;
import java.awt.*;

import static java.awt.EventQueue.invokeLater;

public class PirateWarshipLoginForm extends JFrame {

    private JPanel contentPane;
    private JTextField usernameTextField;
    private JTextField passwordTextField;

    public static void main(String[] args) {
        try {
            UIManager.setLookAndFeel(UIManager.getSystemLookAndFeelClassName());
        } catch (Exception e) {
            e.printStackTrace();
        }

        invokeLater(() -> {
            new PirateWarshipLoginForm();
        });
    }

    private boolean validateUser(String username, String password) {
        return username.equals("user") && password.equals("user");
    }

    /**
     * Create the frame.
     */
    public PirateWarshipLoginForm() {
        setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
        setVisible(true);
        setTitle("Pirate Warships");
        setBounds(new Rectangle(0, 0, 640, 400));
        setResizable(false);
        setLocationRelativeTo(null);

        contentPane = new JPanel();
        contentPane.setBounds(new Rectangle(0, 0, 640, 400));
        contentPane.setBorder(new EmptyBorder(5, 5, 5, 5));
        contentPane.setLayout(null);

        JButton loginBtn = new JButton("Login");
        loginBtn.addActionListener(e -> {
            String username = usernameTextField.getText();
            String password = passwordTextField.getText();

            if (validateUser(username, password)) {
                JOptionPane.showMessageDialog(contentPane, "Login successful.", "Information", JOptionPane.INFORMATION_MESSAGE);
                System.exit(0);
            } else {
                JOptionPane.showMessageDialog(contentPane, "Wrong username or password.", "Error", JOptionPane.ERROR_MESSAGE);
                Thread.interrupted();
            }
        });
        loginBtn.setBorder(new LineBorder(new Color(0, 0, 255), 1, true));
        loginBtn.setBounds(10, 211, 115, 23);
        contentPane.add(loginBtn);

        JButton registrationBtn = new JButton("Registration");
        registrationBtn.setBorder(new LineBorder(new Color(0, 0, 255), 1, true));
        registrationBtn.addActionListener(e -> {
            JOptionPane.showMessageDialog(contentPane, "Registration screen not available.", "Information", JOptionPane.INFORMATION_MESSAGE);
            Thread.interrupted();
        });
        registrationBtn.setBounds(10, 247, 115, 23);
        contentPane.add(registrationBtn);

        usernameTextField = new JTextField();
        usernameTextField.setFont(new Font("Arial Black", Font.BOLD, 11));
        usernameTextField.setOpaque(false);
        usernameTextField.setBorder(new LineBorder(new Color(0, 0, 0), 1, true));
        usernameTextField.setBounds(10, 91, 115, 20);
        contentPane.add(usernameTextField);
        usernameTextField.setColumns(10);

        JLabel lblUser = new JLabel("User");
        lblUser.setFont(new Font("Ariel", Font.BOLD, 14));
        lblUser.setText("<html><font color='black'>User:</font></html>");
        lblUser.setBounds(10, 66, 46, 14);

        contentPane.add(lblUser);

        passwordTextField = new JTextField();
        passwordTextField.setFont(new Font("Arial Black", Font.BOLD, 11));
        passwordTextField.setOpaque(false);
        passwordTextField.setBorder(new LineBorder(new Color(0, 0, 0), 1, true));
        passwordTextField.setBounds(10, 137, 115, 20);
        passwordTextField.setColumns(10);

        contentPane.add(passwordTextField);

        JLabel lblPassword = new JLabel("");
        lblPassword.setFont(new Font("Ariel", Font.BOLD, 14));
        lblPassword.setText("<html><font color='black'>" + "Password:" + "</font></html>");
        lblPassword.setBounds(10, 113, 80, 23);

        contentPane.add(lblPassword);

        JLabel background = new JLabel("");
        background.setHorizontalTextPosition(SwingConstants.CENTER);
        background.setHorizontalAlignment(SwingConstants.CENTER);
        background.setBounds(0, 0, 640, 400);
        ImageIcon icon = new ImageIcon(PirateWarshipLoginForm.class.getResource("../res/PirateShip.gif"));
        background.setIcon(icon);

        contentPane.add(background);

        setContentPane(contentPane);
    }
}

File: src/kotlindemo/PirateWarshipLoginForm.kt
package kotlindemo

import dsl.*
import java.awt.Color
import java.awt.EventQueue.invokeLater
import java.lang.Exception
import javax.swing.*
import javax.swing.border.EmptyBorder
import javax.swing.border.LineBorder


class PirateWarshipLoginForm : JFrame() {

    private lateinit var usernameTextField: JTextField
    private lateinit var passwordTextField: JTextField
    private val backgroundImage by lazy {
        ImageIcon(resource<PirateWarshipLoginForm>("../res/PirateShip.gif"))
    }

    private fun validateUser(username: String, password: String): Boolean {
        return username == "user" && password == "user"
    }

    /**
     * Create the frame.
     */
    init {
        frame(title = "Pirate Warships") {
            defaultCloseOperation = JFrame.EXIT_ON_CLOSE
            bounds = rectangle(dimension = 640 x 400)
            isResizable = false
            isVisible = true
            setLocationRelativeTo(null)

            panel {
                bounds = rectangle(dimension = 640 x 400)
                border = EmptyBorder(5, 5, 5, 5)
                layout = null

                button("Login") {
                    border = LineBorder(Color(0, 0, 255), 1, true)
                    setBounds(10, 211, 115, 23)
                    addActionListener {
                        tryToLogin()
                    }
                }

                button("Registration") {
                    border = LineBorder(Color(0, 0, 255), 1, true)
                    setBounds(10, 247, 115, 23)
                    addActionListener {
                        JOptionPane.showMessageDialog(this@panel, "Registration screen not available.", "Information", JOptionPane.INFORMATION_MESSAGE)
                        Thread.interrupted()
                    }
                }

                usernameTextField = textfield {
                    font = boldFont("Arial Black", 11)
                    isOpaque = false
                    border = LineBorder(Color(0, 0, 0), 1, true)
                    bounds = rectangle(10, 91, 115 x 20)
                    columns = 10
                }

                label("User") {
                    font = boldFont("Ariel")
                    text = blackText("User:")
                    bounds = rectangle(10, 66, 46 x 14)
                }

                passwordTextField = textfield {
                    font = boldFont("Arial Black", 11)
                    isOpaque = false
                    border = LineBorder(Color(0, 0, 0), 1, true)
                    bounds = rectangle(10, 137, 115 x 20)
                    columns = 10
                }

                label("Password") {
                    font = boldFont("Ariel")
                    text = blackText("Password:")
                    bounds = rectangle(10, 113, 80 x 23)
                }

                label {
                    horizontalTextPosition = SwingConstants.CENTER
                    horizontalAlignment = SwingConstants.CENTER
                    bounds = rectangle(dimension = 640 x 400)
                    icon = backgroundImage
                }
            }
        }
    }

    companion object {

        @JvmStatic
        fun main(args: Array<String>) {
            tryWith {
                UIManager.setLookAndFeel(UIManager.getSystemLookAndFeelClassName())
            }

            invokeLater {
                PirateWarshipLoginForm()
            }
        }

        private inline fun tryWith(function: () -> Unit) {
            try {
                function()
            } catch (exception: Exception) {
                exception.printStackTrace()
            }
        }
    }

    private fun JPanel.tryToLogin() {
        val username = usernameTextField.text
        val password = passwordTextField.text

        if (validateUser(username, password)) {
            JOptionPane.showMessageDialog(this, "Login successful.", "Information", JOptionPane.INFORMATION_MESSAGE)
            System.exit(0)
        } else {
            JOptionPane.showMessageDialog(this, "Wrong username or password.", "Error", JOptionPane.ERROR_MESSAGE)
            Thread.interrupted()
        }
    }
}
